opengrok-indexer/src/main/java/org/opengrok/indexer/index/Indexer.java

```diff
@@ -161,9 +163,36 @@ public static void main(String[] argv) {
 
         boolean createDict = false;
 
+        int CONNECT_TIMEOUT = 1000;  // in milliseconds
+
         try {
             argv = parseOptions(argv);
 
+            if (webappURI != null) {
+                boolean connectWorks = false;
+                try {
+                    for (InetAddress addr : InetAddress.getAllByName(HostUtil.urlToHostname(webappURI))) {
+                        int port = HostUtil.urlToPort(webappURI);
+                        if (port <= 0) {
+                            LOGGER.log(Level.SEVERE, "invalid port number for " + webappURI);
+                            break;
+                        }
+                        if (HostUtil.isReachable(addr, port, CONNECT_TIMEOUT)) {
+                            LOGGER.log(Level.FINE, "URI " + webappURI + " is reachable via " + addr.toString());
+                            connectWorks = true;
+                            break;
+                        }
+                    }
+                } catch (URISyntaxException e) {
+                    LOGGER.log(Level.WARNING, String.format("URI not valid: %s", webappURI), e);
+                }
+
+                if (!connectWorks) {
+                    LOGGER.log(Level.SEVERE, webappURI + " is not reachable.");
```

Comment:
It seems we're printing the message to `System.err` right before the exit in other places. Will the log be always displayed?